src/vm/i386/excepcpu.h

```diff
@@ -28,12 +28,37 @@ class Thread;
                               // Actually, the handler getting set is properly registered
 #endif
 
+#ifdef FEATURE_PAL
+
+extern VOID SetSEHRecord(PEXCEPTION_REGISTRATION_RECORD record);
```

Comment:
I once tried that, but failed. Clang recognizes ``__readfsdword`` as a builtin function. 

The implementation of ``SetSEHRecord`` and ``ResetSEHRecord`` is almost same as you suggested (for ``__writefdsword`` and ``__readfsdword``). I just put them in excepx86.cpp as GetCurrentSEHRecord need to revised anyway (it is called inside VerifyValidTransitionFromManagedCode, and incurs segmentation fault at __readfsdword while trying to access %fs segment).